syncer/causality.go

```diff
@@ -14,31 +14,112 @@
 package syncer
 
 import (
-	"github.com/pingcap/dm/pkg/terror"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pingcap/dm/pkg/log"
+	"github.com/pingcap/dm/syncer/metrics"
 )
 
-// causality provides a simple mechanism to improve the concurrency of SQLs execution under the premise of ensuring correctness.
+// Causality provides a simple mechanism to improve the concurrency of SQLs execution under the premise of ensuring correctness.
 // causality groups sqls that maybe contain causal relationships, and syncer executes them linearly.
-// if some conflicts exist in more than one groups, then syncer waits all SQLs that are grouped be executed and reset causality.
+// if some conflicts exist in more than one groups, causality generate a conflict job and reset.
 // this mechanism meets quiescent consistency to ensure correctness.
-type causality struct {
-	relations map[string]string
+type Causality struct {
```

Comment:
seems this structure is not needed to be public?

Reply:
use lowwer case 13bed05